snuba/query/conditions.py

```diff
@@ -321,3 +321,23 @@ def build_match(
             ),
         ]
     )
+
+
+def get_literals_for_column_condition(
+    col: str, exp_type: Any, conditions: Expression
+) -> Sequence[Literal]:
+    match = build_match(col, (ConditionFunctions.EQ,), exp_type)
+    for exp in conditions:
+        result = match.match(exp)
```

Comment:
This seems a very specific use case. The idea of scanning the whole condition tree to find a binary condition (independently whether it is in a nAND or an OR) seems svery specific to your use case. I would not put it in the general query api. I would instead add it where you use it.